python313Packages.django: 4.2.x -> 5.2.x (bugfixes)

mweinelt opened the pull request:
<del>Bump the default django to django_5.</del>

Fix regressions caused by bumping default django to 5.x.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mweinelt commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 396590`

---
### `x86_64-linux`
<details>
  <summary>:x: 7 packages failed to build:</summary>
  <ul>
    <li>paperless-ngx</li>
    <li>pretalx</li>
    <li>pretalx.dist</li>
    <li>pretalx.static</li>
    <li>weblate</li>
    <li>weblate.dist</li>
    <li>weblate.static</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 48 packages built:</summary>
  <ul>
    <li>glitchtip</li>
    <li>healthchecks</li>
    <li>netbox (netbox_4_2)</li>
    <li>netbox_4_1</li>
    <li>python312Packages.django-apscheduler</li>
    <li>python312Packages.django-apscheduler.dist</li>
    <li>python312Packages.django_5</li>
    <li>python312Packages.django_5.dist</li>
    <li>python312Packages.netbox-bgp</li>
    <li>python312Packages.netbox-bgp.dist</li>
    <li>python312Packages.netbox-contract</li>
    <li>python312Packages.netbox-contract.dist</li>
    <li>python312Packages.netbox-documents</li>
    <li>python312Packages.netbox-documents.dist</li>
    <li>python312Packages.netbox-floorplan-plugin</li>
    <li>python312Packages.netbox-floorplan-plugin.dist</li>
    <li>python312Packages.netbox-interface-synchronization</li>
    <li>python312Packages.netbox-interface-synchronization.dist</li>
    <li>python312Packages.netbox-napalm-plugin</li>
    <li>python312Packages.netbox-napalm-plugin.dist</li>
    <li>python312Packages.netbox-plugin-prometheus-sd</li>
    <li>python312Packages.netbox-plugin-prometheus-sd.dist</li>
    <li>python312Packages.netbox-qrcode</li>
    <li>python312Packages.netbox-qrcode.dist</li>
    <li>python312Packages.netbox-reorder-rack</li>
    <li>python312Packages.netbox-reorder-rack.dist</li>
    <li>python312Packages.netbox-routing</li>
    <li>python312Packages.netbox-routing.dist</li>
    <li>python312Packages.netbox-topology-views</li>
    <li>python312Packages.netbox-topology-views.dist</li>
    <li>python313Packages.django-apscheduler</li>
    <li>python313Packages.django-apscheduler.dist</li>
    <li>python313Packages.django_5</li>
    <li>python313Packages.django_5.dist</li>
    <li>python313Packages.netbox-bgp</li>
    <li>python313Packages.netbox-bgp.dist</li>
    <li>python313Packages.netbox-documents</li>
    <li>python313Packages.netbox-documents.dist</li>
    <li>python313Packages.netbox-interface-synchronization</li>
    <li>python313Packages.netbox-interface-synchronization.dist</li>
    <li>python313Packages.netbox-plugin-prometheus-sd</li>
    <li>python313Packages.netbox-plugin-prometheus-sd.dist</li>
    <li>python313Packages.netbox-qrcode</li>
    <li>python313Packages.netbox-qrcode.dist</li>
    <li>python313Packages.netbox-reorder-rack</li>
    <li>python313Packages.netbox-reorder-rack.dist</li>
    <li>python313Packages.netbox-routing</li>
    <li>python313Packages.netbox-routing.dist</li>
  </ul>
</details>

mweinelt commented:
Cause are two test failures in crispy-bootstrap3, that I've reported upstream.

https://github.com/django-crispy-forms/crispy-bootstrap3/issues/12

No pressure, we can wait a while for upstream to respond.

mweinelt commented:
@erictapen Note that weblate is the only remaining package affected by the `django-crispy-bootstrap3` build failure.

mweinelt committed:
python3Packages.django-jinja: fix build with django 5.2

Also migrate to PEP517 builder and use pytest for tests.

mweinelt committed:
python3Packages.django-jinja: normalize package directory and cleanup

mweinelt committed:
python3Packages.django-cryptography: 1.1 -> 1.1-unstable-2024-02-16

Upstream author has apparently vanished, so this is probably it until a
fork appears.

mweinelt committed:
python3Packages.django-scheduler: fix build with django 5.2

mweinelt commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 396590`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 5 packages marked as broken and skipped:</summary>
  <ul>
    <li>netbox_4_1</li>
    <li>python312Packages.django-sites</li>
    <li>python312Packages.django-sites.dist</li>
    <li>python313Packages.django-sites</li>
    <li>python313Packages.django-sites.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 13 packages failed to build:</summary>
  <ul>
    <li>glitchtip</li>
    <li>paperless-ngx</li>
    <li>paperless-ngx.dist</li>
    <li>pretalx</li>
    <li>pretalx.dist</li>
    <li>pretalx.static</li>
    <li>python312Packages.django-scheduler</li>
    <li>python312Packages.django-scheduler.dist</li>
    <li>python313Packages.django-scheduler</li>
    <li>python313Packages.django-scheduler.dist</li>
    <li>weblate</li>
    <li>weblate.dist</li>
    <li>weblate.static</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 86 packages built:</summary>
  <ul>
    <li>healthchecks</li>
    <li>netbox (netbox_4_2)</li>
    <li>pretix</li>
    <li>pretix.dist</li>
    <li>python312Packages.django-apscheduler</li>
    <li>python312Packages.django-apscheduler.dist</li>
    <li>python312Packages.django-bootstrap5</li>
    <li>python312Packages.django-bootstrap5.dist</li>
    <li>python312Packages.django-cryptography</li>
    <li>python312Packages.django-cryptography.dist</li>
    <li>python312Packages.django-currentuser</li>
    <li>python312Packages.django-currentuser.dist</li>
    <li>python312Packages.django-jinja</li>
    <li>python312Packages.django-jinja.dist</li>
    <li>python312Packages.django-pattern-library</li>
    <li>python312Packages.django-pattern-library.dist</li>
    <li>python312Packages.django-payments</li>
    <li>python312Packages.django-payments.dist</li>
    <li>python312Packages.django-phonenumber-field</li>
    <li>python312Packages.django-phonenumber-field.dist</li>
    <li>python312Packages.django-two-factor-auth</li>
    <li>python312Packages.django-two-factor-auth.dist</li>
    <li>python312Packages.django_5</li>
    <li>python312Packages.django_5.dist</li>
    <li>python312Packages.drf-extra-fields</li>
    <li>python312Packages.drf-extra-fields.dist</li>
    <li>python312Packages.netbox-attachments</li>
    <li>python312Packages.netbox-attachments.dist</li>
    <li>python312Packages.netbox-bgp</li>
    <li>python312Packages.netbox-bgp.dist</li>
    <li>python312Packages.netbox-contract</li>
    <li>python312Packages.netbox-contract.dist</li>
    <li>python312Packages.netbox-documents</li>
    <li>python312Packages.netbox-documents.dist</li>
    <li>python312Packages.netbox-floorplan-plugin</li>
    <li>python312Packages.netbox-floorplan-plugin.dist</li>
    <li>python312Packages.netbox-interface-synchronization</li>
    <li>python312Packages.netbox-interface-synchronization.dist</li>
    <li>python312Packages.netbox-napalm-plugin</li>
    <li>python312Packages.netbox-napalm-plugin.dist</li>
    <li>python312Packages.netbox-plugin-prometheus-sd</li>
    <li>python312Packages.netbox-plugin-prometheus-sd.dist</li>
    <li>python312Packages.netbox-qrcode</li>
    <li>python312Packages.netbox-qrcode.dist</li>
    <li>python312Packages.netbox-reorder-rack</li>
    <li>python312Packages.netbox-reorder-rack.dist</li>
    <li>python312Packages.netbox-routing</li>
    <li>python312Packages.netbox-routing.dist</li>
    <li>python312Packages.netbox-topology-views</li>
    <li>python312Packages.netbox-topology-views.dist</li>
    <li>python313Packages.django-apscheduler</li>
    <li>python313Packages.django-apscheduler.dist</li>
    <li>python313Packages.django-bootstrap5</li>
    <li>python313Packages.django-bootstrap5.dist</li>
    <li>python313Packages.django-cryptography</li>
    <li>python313Packages.django-cryptography.dist</li>
    <li>python313Packages.django-currentuser</li>
    <li>python313Packages.django-currentuser.dist</li>
    <li>python313Packages.django-jinja</li>
    <li>python313Packages.django-jinja.dist</li>
    <li>python313Packages.django-pattern-library</li>
    <li>python313Packages.django-pattern-library.dist</li>
    <li>python313Packages.django-payments</li>
    <li>python313Packages.django-payments.dist</li>
    <li>python313Packages.django-phonenumber-field</li>
    <li>python313Packages.django-phonenumber-field.dist</li>
    <li>python313Packages.django-two-factor-auth</li>
    <li>python313Packages.django-two-factor-auth.dist</li>
    <li>python313Packages.django_5</li>
    <li>python313Packages.django_5.dist</li>
    <li>python313Packages.drf-extra-fields</li>
    <li>python313Packages.drf-extra-fields.dist</li>
    <li>python313Packages.netbox-bgp</li>
    <li>python313Packages.netbox-bgp.dist</li>
    <li>python313Packages.netbox-documents</li>
    <li>python313Packages.netbox-documents.dist</li>
    <li>python313Packages.netbox-interface-synchronization</li>
    <li>python313Packages.netbox-interface-synchronization.dist</li>
    <li>python313Packages.netbox-plugin-prometheus-sd</li>
    <li>python313Packages.netbox-plugin-prometheus-sd.dist</li>
    <li>python313Packages.netbox-qrcode</li>
    <li>python313Packages.netbox-qrcode.dist</li>
    <li>python313Packages.netbox-reorder-rack</li>
    <li>python313Packages.netbox-reorder-rack.dist</li>
    <li>python313Packages.netbox-routing</li>
    <li>python313Packages.netbox-routing.dist</li>
  </ul>
</details>

erictapen commented:
> @erictapen Note that weblate is the only remaining package affected by the `django-crispy-bootstrap3` build failure.

I won't have the time to look into this and will need to wait for a solution from upstream. Are you planning to include this in the release?

mweinelt commented:
No, this is 25.11 material.

SuperSandro2000 commented on pkgs/development/python-modules/django-jinja/default.nix:
```diff
@@ -3,37 +3,61 @@
   pname = "django-jinja";
   version = "2.11.0";
+  pyproject = true;
 
   disabled = pythonOlder "3.8";
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/django-jinja/default.nix:
```diff
@@ -3,37 +3,61 @@
 
+  build-system = [ setuptools ];
+
   propagatedBuildInputs = [
```
```suggestion
  dependencies = [
```

SuperSandro2000 reviewed (commented):
Do we want to replace django_5_2 with django_5 as in glitchtip, netbox or paperless-ngx?

mweinelt committed:
mailman: pin to django 4.2

mweinelt committed:
python3Packages.django-rest-registration: fix build

Wrong jwt dependency broke a test. Accelerate the otherwise slow tests
with xdist and clean up and reannotate failing tests.

mweinelt committed:
python3Packages.mezzanine: mark broken with django>=5

mweinelt committed:
python3Packages.pylint-django: 2.6.1 -> 2.6.1-unstable-2025-11-09

Fast-forward for Django 5.2 support.

mweinelt committed:
python3Packages.pyscaffoldext-django: relax django, modernize, cleanup
